### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by yosriz**
add integration tests on real horizon test net
* make wipeout method work (add clearAllAccount in KeyStore) for integration tests
* fix some bugs with KeyStore 
* mark all network dependant long running tests as @LargeTest
* skip those tests in travis build

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -40,5 +40,7 @@ before_script:
 script:
   - ./gradlew :sample:assembleDebug
   - ./gradlew :kin-sdk-core:test
-  - ./gradlew :kin-sdk-core:connectedAndroidTest
+  - ./gradlew kin-sdk-core:assembleDebugAndroidTest
+  - adb install -r ./kin-sdk-core/build/outputs/apk/androidTest/debug/kin-sdk-core-debug-androidTest.apk
+  - adb shell am instrument -w -e notAnnotation android.support.test.filters.LargeTest  kin.sdk.core.test/android.support.test.runner.AndroidJUnitRunner
```
Comment (OrenZak) [inline .travis.yml]: add comments?

File: dependencies.gradle [hunk 2]
```diff
@@ -16,19 +17,21 @@ ext {
...
             junit         : buildDependency(junitPackage, 'junit', junitVersion),
-            mockitoAndroid: buildDependency(mockitoPackage,'mockito-android', mockitoAndroidVersion),
+            mockitoAndroid: buildDependency(mockitoPackage, 'mockito-android', mockitoAndroidVersion),
             mockito       : buildDependency(mockitoPackage, 'mockito-core', mockitoVersion),
             androidTestRunner : buildDependency(androidTestPackage, 'runner', androidTestVersion),
             androidTestRules : buildDependency(androidTestPackage, 'rules', androidTestVersion),
             mockWebServer : buildDependency(okHttpPackage, 'mockwebserver', mockWebServerVersion),
-            robolectric   : buildDependency(robolectricPackage, 'robolectric', robolectricVersion)
+            robolectric   : buildDependency(robolectricPackage, 'robolectric', robolectricVersion),
+            hamcrest   : buildDependency(hamcrestPackage, 'hamcrest-library', hamcrestVersion)
```
Comment (OrenZak) [inline dependencies.gradle]: format

File: kin-sdk-core/build.gradle [hunk 3]
```diff
@@ -31,7 +31,7 @@ android {
 }
 
 dependencies {
-    implementation 'com.github.kinfoundation:android-stellar-sdk:master-SNAPSHOT'
+    implementation 'com.github.kinfoundation:android-stellar-sdk:support_below_api19-SNAPSHOT'
```
Comment (OrenZak) [inline kin-sdk-core/build.gradle]: move to `dependencies.gradle`

File: kin-sdk-core/src/androidTest/java/kin/sdk/core/FakeKinIssuer.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import org.stellar.sdk.responses.AccountResponse;
+import org.stellar.sdk.responses.SubmitTransactionResponse;
+
+/**
+ * Fake issuer for integration test, support creating and funding accounts on stellar test net
+ */
+class FakeKinIssuer {
+
+    static final String KIN_ISSUER_ACCOUNT_ID = "GBA2XHZRUAHEL4DZX7XNHR7HLBAUYPRNKLD2PIUKWV2LVVE6OJT4NDLM";
+    private static final String KIN_ISSUER_SEED = "SDL34EFTWFPIVJLUYSCVZJFYPYLIGFHDYG5VKOVBQYLU6H4OWRPETCAE";
```
Comment (OrenZak) [inline kin-sdk-core/src/androidTest/java/kin/sdk/core/FakeKinIssuer.java]: let's generate it in the code every time.

File: kin-sdk-core/src/androidTest/java/kin/sdk/core/FakeKinIssuer.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    private static final String TEST_NETWORK_URL = "https://horizon-testnet.stellar.org";
```
Comment (OrenZak) [inline kin-sdk-core/src/androidTest/java/kin/sdk/core/FakeKinIssuer.java]: we can divide to scheme and base path

File: kin-sdk-core/src/androidTest/java/kin/sdk/core/FakeKinIssuer.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        this.server = new Server(TEST_NETWORK_URL, TIMEOUT_SEC, TimeUnit.SECONDS);
+        this.issuerKeyPair = KeyPair.fromSecretSeed(KIN_ISSUER_SEED);
+        this.asset = Asset.createNonNativeAsset("KIN", issuerKeyPair);
+        createAndFundWithLumens(KIN_ISSUER_ACCOUNT_ID);
+    }
+
+    private void createAndFundWithLumens(String accountId) throws IOException {
+        HttpUrl url = new HttpUrl.Builder()
+            .scheme("https")
+            .host("horizon-testnet.stellar.org")
```
Comment (OrenZak) [inline kin-sdk-core/src/androidTest/java/kin/sdk/core/FakeKinIssuer.java]: use the consts

File: kin-sdk-core/src/androidTest/java/kin/sdk/core/FakeKinIssuer.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            .addPathSegment("friendbot")
+            .addQueryParameter("addr", accountId)
+            .build();
+        OkHttpClient client = new OkHttpClient();
+        okhttp3.Request request = new okhttp3.Request.Builder().url(url).build();
+        Response response = client.newCall(request).execute();
+        assertTrue(response != null && response.body() != null);
```
Comment (OrenZak) [inline kin-sdk-core/src/androidTest/java/kin/sdk/core/FakeKinIssuer.java]: check error code

File: kin-sdk-core/src/androidTest/java/kin/sdk/core/FakeKinIssuer.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+class FakeKinIssuer {
+
+    static final String KIN_ISSUER_ACCOUNT_ID = "GBA2XHZRUAHEL4DZX7XNHR7HLBAUYPRNKLD2PIUKWV2LVVE6OJT4NDLM";
+    private static final String KIN_ISSUER_SEED = "SDL34EFTWFPIVJLUYSCVZJFYPYLIGFHDYG5VKOVBQYLU6H4OWRPETCAE";
+    private static final String TEST_NETWORK_URL = "https://horizon-testnet.stellar.org";
+    private static final int TIMEOUT_SEC = 20;
+
+    private final Server server;
+    private final KeyPair issuerKeyPair;
+    private final Asset asset;
```
Comment (OrenZak) [inline kin-sdk-core/src/androidTest/java/kin/sdk/core/FakeKinIssuer.java]: rename to kinAsset

File: kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java [hunk 9]
```diff
@@ -20,85 +23,142 @@
...
+        deleteAccount(0, passphrase);
+    }
+
+    /**
+     * Deletes the account at input index (if it exists)
+     *
+     * @param passphrase the passphrase used when the account was created
+     */
+    public void deleteAccount(int index, String passphrase) throws DeleteAccountException {
+        if (getAccountsCount() > index) {
```
Comment (OrenZak) [inline kin-sdk-core/src/main/java/kin/sdk/core/KinClient.java]: check index < 0

File: kin-sdk-core/src/test/java/kin/sdk/core/KinClientTest.java [hunk 10]
```diff
@@ -25,49 +22,117 @@
...
-
-        kinClient = new KinClient(clientWrapper);
+        when(mockClientWrapper.getKeyStore()).thenAnswer(invocation -> fakeKeyStore);
     }
 
     @Test
     public void createAccount_NewAccount() throws Exception {
         fakeKeyStore = new FakeKeyStore();
-
+        kinClient = new KinClient(mockClientWrapper);
```
Comment (OrenZak) [inline kin-sdk-core/src/test/java/kin/sdk/core/KinClientTest.java]: check if we can init in setUp

File: kin-sdk-core/src/test/java/kin/sdk/core/KinClientTest.java [hunk 11]
```diff
@@ -25,49 +22,117 @@
...
-    public void createAccount_Exception() throws Exception {
-        String expectedExString = "My Exception";
-        fakeKeyStore = mock(KeyStore.class);
-        when(fakeKeyStore.newAccount(anyString())).thenThrow(new RuntimeException(expectedExString));
-        expectedEx.expect(CreateAccountException.class);
-        expectedEx.expectCause(new HasPropertyWithValue<>("message", is(expectedExString)));
+    public void createAccount_AddAccount() throws Exception {
+        fakeKeyStore = new FakeKeyStore();
+        kinClient = new KinClient(mockClientWrapper);
+        KinAccount kinAccount = kinClient.addAccount("123456");
```
Comment (OrenZak) [inline kin-sdk-core/src/test/java/kin/sdk/core/KinClientTest.java]: use `PASSPHRASE` const

File: kin-sdk-core/src/test/java/kin/sdk/core/KinClientTest.java [hunk 12]
```diff
@@ -25,49 +22,117 @@
...
+        assertNotNull(expectedAccount2);
+        assertThat(account1.getAccountId(), equalTo(expectedAccount1.getPublicAddress()));
+        assertThat(account2.getAccountId(), equalTo(expectedAccount2.getPublicAddress()));
     }
 
     @Test
     public void createAccount_ExistingAccount_SameAccount() throws Exception {
         Account account = createKeyStoreWithRandomAccount();
 
+        kinClient = new KinClient(mockClientWrapper);
```
Comment (OrenZak) [inline kin-sdk-core/src/test/java/kin/sdk/core/KinClientTest.java]: please check also for the count.

File: kin-sdk-core/src/androidTest/java/kin/sdk/core/KinAccountIntegrationTest.java [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+    static public void setupKinIssuer() throws IOException {
+        fakeKinIssuer = new FakeKinIssuer();
+    }
+
+    @Before
+    public void setup() throws IOException {
+        ServiceProvider serviceProvider = new ServiceProvider(TEST_NETWORK_URL, FakeKinIssuer.KIN_ISSUER_ACCOUNT_ID);
+        kinClient = new KinClient(InstrumentationRegistry.getTargetContext(), serviceProvider);
+        kinClient.wipeoutAccount();
+    }
```
Comment (OrenZak) [inline kin-sdk-core/src/androidTest/java/kin/sdk/core/KinAccountIntegrationTest.java]: Maybe add `@after` and wipeoutAccount

**Review by OrenZak — CHANGES_REQUESTED** [review]
Good job !!!!! see the comments :)

**Comment by yosriz** [comment]
@OrenZak Thanks, changes applied, also for https://github.com/kinfoundation/kin-sdk-core-stellar-android/pull/13, please approve :smile:

**Comment by OrenZak** [comment]
👍 Great work !! thanks !

**Review by OrenZak — APPROVED** [review]

**Pull request merged**